### DISCUSSION TITLE
How to find difference between two python file?

### DISCUSSION THREAD

**Discussion by hetsolanki97** [GumTreeDiff/gumtree#312] [Q&A]
Is there an equivalent functionality in the Python parser, similar to the one available in Java, that allows us to find the differences between two Python files?
Note: We have used the Python parser and are able to find the XML for both files. In addition, we are also able to extract the metadata from the XML. We are trying to use XML provided by Python in the Java code and find the difference. Is this the correct approach?

**Comment by jrfaller** [comment]
Hi!

Python is in our list of supported languages, there are two parsers for Python that you can try.

**Reply by hetsolanki97** [reply]
Hi!

Thanks for your reply. We used the Python parser and were able to get the output of the entire gum tree. Is there a way to get information for a specific node like update or delete using any function?

**Comment by jrfaller** [comment]
Normally you can diff python ASTs like anything else vie the `webdiff` command or the Java API.

**Reply by hetsolanki97** [reply]
Thank you so much for your prompt response. :) 
Correct me if wrong but Java API works for Java code, right?
While we use the webdiff command it gives us output in the web browser how to get it into our program in the console?

**Comment by jrfaller** [comment]
Hi,

From the Java API you can diff code in any language GumTree is compatible with (see unit tests in the gen.* modules).
